x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/components/package_policy_input_var_field.tsx

```diff
@@ -296,6 +316,61 @@ function getInputComponent({
   }
 }
 
+const SecretFieldWrapper = ({ children }: { children: React.ReactNode }) => {
+  return (
+    <EuiPanel hasShadow={false} color="subdued" paddingSize="s">
+      {children}
+    </EuiPanel>
+  );
+};
+
+const SecretFieldLabel = ({ fieldLabel }: { fieldLabel: string }) => {
+  const { docLinks } = useStartServices();
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const onButtonClick = () => setIsPopoverOpen(!isPopoverOpen);
+  const closePopover = () => setIsPopoverOpen(false);
+
+  return (
+    <EuiFlexGroup alignItems="center" gutterSize="xs">
+      <EuiFlexItem grow={true} aria-label={fieldLabel}>
+        {fieldLabel}
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiPopover
+          isOpen={isPopoverOpen}
+          button={<EuiButtonIcon size="s" iconType="lock" onClick={onButtonClick} />}
+          closePopover={closePopover}
+          anchorPosition="upLeft"
+        >
+          <EuiText size="s">
+            <p
+              css={`
+                max-width: 400px;
+              `}
+            >
+              <FormattedMessage
+                id="xpack.fleet.createPackagePolicy.stepConfigure.secretLearnMorePopoverContent"
+                defaultMessage="This value is a secret. Once you save this integration policy, you won't be able to view the value again. {learnMoreLink}."
```

Comment:
```suggestion
                defaultMessage="This value is a secret. After you save this integration policy, you won't be able to view the value again. {learnMoreLink}."
```
("once" can be kind of a confusing term for non-native English speakers.)